cmd/proxy/main.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"time"
+
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/acmpca"
+	"github.com/cloudflare/cfssl/auth"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/sirupsen/logrus"
+	logrusadapter "logur.dev/adapter/logrus"
+
+	"github.com/johanbrandhorst/certify"
+	"github.com/johanbrandhorst/certify/cmd/proxy/internal/envtypes"
+	"github.com/johanbrandhorst/certify/issuers/aws"
+	"github.com/johanbrandhorst/certify/issuers/cfssl"
+	"github.com/johanbrandhorst/certify/issuers/vault"
+)
+
+type configuration struct {
+	Addr      string             `default:":443" desc:"The address to serve the proxy on."`
+	ProxyAddr string             `default:"localhost:80" split_words:"true" desc:"The host to proxy to, without scheme."`
+	LogFormat envtypes.LogFormat `default:"json" split_words:"true" desc:"Log formatter to use. One of json or text."`
+
+	Issuer envtypes.Issuer `required:"true" desc:"The certificate issuer to use. One of vault, cfssl or aws."`
+	Vault  envtypes.Vault
+	CFSSL  envtypes.CFSSL
+	AWS    envtypes.AWS
+
+	CacheDir                  string                `split_words:"true" desc:"Directory to cache certs. If unset, will cache in memory."`
+	CommonName                string                `required:"true" split_words:"true" desc:"The Common Name that will be used when issuing certificates. This can be a DNS record or a regular name."`
+	RenewBefore               time.Duration         `split_words:"true" default:"4h" desc:"How long before expiry a certificate should be considered too old to use when fetched from the cache."`
+	IssueTimeout              time.Duration         `default:"1m" split_words:"true"  desc:"The upper bound of time allowed per Issue request."`
+	SubjectAlternativeNames   []string              `split_words:"true" desc:"A comma-separated list of DNS names that should be included in the SANs of the issued certificates."`
+	IPSubjectAlternativeNames []net.IP              `envconfig:"IP_SUBJECT_ALTERNATIVE_NAMES" desc:"A comma-separated list of IPs that should be included in the IPSANs of the issued certificates."`
+	KeyGenerator              envtypes.KeyGenerator `split_words:"true" default:"ecdsa" desc:"The key algorithm to use for new certificates. One of ecdsa and rsa."`
+}
+
+func main() {
+	conf := configuration{}
+	err := envconfig.Process("", &conf)
+	if err != nil {
+		_ = envconfig.Usage("", &conf)
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	log := logrus.New()
+	log.Formatter = conf.LogFormat
+
+	var issuer certify.Issuer
+	switch conf.Issuer {
```

Comment:
I would move this to a function to keep the main a bit more compact, but that's just a style thing...